pkgs/development/compilers/rust/1_49.nix

```diff
@@ -39,9 +39,11 @@ import ./default.nix {
   bootstrapHashes = {
     i686-unknown-linux-gnu = "7fdb8836a1f0427d5b47e6a2d496f67ebff04350407411f57cf20c9b3544e26f";
     x86_64-unknown-linux-gnu = "950420a35b2dd9091f1b93a9ccd5abc026ca7112e667f246b1deb79204e2038b";
+    x86_64-unknown-linux-musl = "4ed9627f57b4e0b9807fc5e7513d9731f4791668b7f875b9e44e65e21072c56f";
```

Comment:
Last time I checked musl binaries also require musl when doing patchelf. Did this work for you?